### ISSUE TITLE
Fresh install: PHP Fatal error:  Class 'Log' not found in /var/www/snipe-it/app/Exceptions/Handler.php on line 39

### ISSUE THREAD

**Issue by m384wwn** [snipe/snipe-it#4470]
Getting client side HTTP ERROR 500 when accessing snipe-it via web for the first time to complete Pre-Flight & Setup.

Server side:
PHP Fatal error:  Class 'Log' not found in /var/www/snipe-it/app/Exceptions/Handler.php on line 39

-----

#### Please confirm you have done the following before posting your bug report:

- [X] I have enabled debug mode 
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Provide answers to these questions:

- Is this a fresh install or an upgrade? FRESH
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) MANUAL

[root@web05 snipe-it]#  git describe --tags
v4.1.4-5-g4494456

[root@web05 snipe-it]# php --version
PHP 5.6.32 (cli) (built: Oct 29 2017 19:00:01) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies

[root@web05 snipe-it]# cat /etc/redhat-release 
CentOS release 6.9 (Final)

[root@web05 snipe-it]# httpd -v
Server version: Apache/2.2.15 (Unix)
Server built:   Oct 19 2017 16:43:38

[root@mysql01 ~]# mysql -V
mysql  Ver 14.14 Distrib 5.1.73, for redhat-linux-gnu (x86_64) using readline 5.1

[root@web05 snipe-it]# composer -v
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Composer version 1.5.2 2017-09-11 16:59:25

[Composer was global installed, this VM only has snipe-it so there are no other users than root/(apache).]

Composer dependencies are installed:
[root@web05 snipe-it]# composer install --no-dev --prefer-source
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Loading composer repositories with package information
Installing dependencies from lock file
Nothing to install or update
Generating optimized autoload files

My .env file: 
```
# --------------------------------------------
# REQUIRED: BASIC APP SETTINGS
# --------------------------------------------
APP_ENV=production
APP_DEBUG=true
APP_KEY=base64:XXXXXXXXXXXXXXXXXXXXXXX=
APP_URL=https://asset.domain.tld
APP_TIMEZONE='America/Toronto'
APP_LOCALE=en

# --------------------------------------------
# REQUIRED: DATABASE SETTINGS
# --------------------------------------------
DB_CONNECTION=mysql
DB_HOST=mysql01.domain.tld
DB_DATABASE=db_snipeit
DB_USERNAME=svc_snipeit
DB_PASSWORD=XXXXXXXXXXXX
DB_PREFIX=null
DB_DUMP_PATH='/usr/bin'
DB_CHARSET=utf8mb4
DB_COLLATION=utf8mb4_unicode_ci

# --------------------------------------------
# OPTIONAL: SSL DATABASE SETTINGS
# --------------------------------------------
DB_SSL=false
DB_SSL_KEY_PATH=null
DB_SSL_CERT_PATH=null
DB_SSL_CA_PATH=null
DB_SSL_CIPHER=null

# --------------------------------------------
# REQUIRED: OUTGOING MAIL SERVER SETTINGS
# --------------------------------------------
MAIL_DRIVER=smtp
MAIL_HOST=smtp-relay.gmail.com
MAIL_PORT=587
MAIL_USERNAME=YOURUSERNAME
MAIL_PASSWORD=YOURPASSWORD
MAIL_ENCRYPTION=tls
MAIL_FROM_ADDR=snipe-it@domain.tld
MAIL_FROM_NAME='Snipe-IT'
MAIL_REPLYTO_ADDR=snipe-it@domain.tld
MAIL_REPLYTO_NAME='Snipe-IT'

# --------------------------------------------
# REQUIRED: IMAGE LIBRARY
# This should be gd or imagick
# --------------------------------------------
IMAGE_LIB=gd
# --------------------------------------------
# OPTIONAL: SESSION SETTINGS
# --------------------------------------------
SESSION_LIFETIME=12000
EXPIRE_ON_CLOSE=false
ENCRYPT=false
COOKIE_NAME=snipeit_session
COOKIE_DOMAIN=null
SECURE_COOKIES=false

# --------------------------------------------
# OPTIONAL: SECURITY HEADER SETTINGS
# --------------------------------------------
REFERRER_POLICY=same-origin
ENABLE_CSP=false

# --------------------------------------------
# OPTIONAL: CACHE SETTINGS
# --------------------------------------------
CACHE_DRIVER=file
SESSION_DRIVER=file
QUEUE_DRIVER=sync

# --------------------------------------------
# OPTIONAL: REDIS SETTINGS
# --------------------------------------------
REDIS_HOST=null
REDIS_PASSWORD=null
REDIS_PORT-null

# --------------------------------------------
# OPTIONAL: AWS S3 SETTINGS
# --------------------------------------------
AWS_SECRET=null
AWS_KEY=null
AWS_REGION=null
AWS_BUCKET=null

# --------------------------------------------
# OPTIONAL: LOGIN THROTTLING
# --------------------------------------------
LOGIN_MAX_ATTEMPTS=5
LOGIN_LOCKOUT_DURATION=60

# --------------------------------------------
# OPTIONAL: MISC
# --------------------------------------------
APP_LOG=single
APP_LOG_MAX_FILES=10
APP_LOCKED=false
FILESYSTEM_DISK=local
APP_TRUSTED_PROXIES=192.168.1.1,10.0.0.1
ALLOW_IFRAMING=false
APP_CIPHER=AES-256-CBC
```

My directory:
```
[root@web05 snipe-it]# ll -a
total 528
drwxr-xr-x 17 apache apache   4096 Nov 16 14:31 .
drwxr-xr-x  7 root   root     4096 Nov 15 15:04 ..
-rw-r--r--  1 apache apache  21549 Nov 15 15:05 .all-contributorsrc
d-wx--x--x 17 apache apache   4096 Nov 15 15:05 app
--w-------  1 apache apache   1646 Nov 15 15:05 artisan
d-wx--x--x  3 apache apache   4096 Nov 15 15:05 bootstrap
d-wx--x--x 22 apache apache   4096 Nov 15 15:05 bower_components
--w-------  1 apache apache    875 Nov 15 15:05 bower.json
--w-------  1 apache apache   9210 Nov 15 15:05 c3.php
--w-------  1 apache apache    373 Nov 15 15:05 codeception.yml
--w-------  1 apache apache   3228 Nov 15 15:05 CODE_OF_CONDUCT.md
--w-------  1 apache apache   2273 Nov 15 15:05 composer.json
--w-------  1 apache apache 224703 Nov 15 15:05 composer.lock
d-wx--x--x  2 apache apache   4096 Nov 15 15:05 config
--w-------  1 apache apache    317 Nov 15 15:05 CONTRIBUTING.md
--w-------  1 apache apache     90 Nov 15 15:05 crowdin.yml
d-wx--x--x  5 apache apache   4096 Nov 15 15:05 database
d-wx--x--x  2 apache apache   4096 Nov 16 14:10 docker
--w-------  1 apache apache    667 Nov 15 15:05 docker.env
--w-------  1 apache apache   2898 Nov 15 15:05 Dockerfile
-rw-r--r--  1 apache apache     67 Nov 15 15:05 .dockerignore
-rw-r--r--  1 apache apache   2863 Nov 16 14:18 .env
-rw-r--r--  1 apache apache   2731 Nov 15 15:05 .env.example
-rw-r--r--  1 apache apache   1826 Nov 15 15:05 .env.testing
-rw-r--r--  1 apache apache   1755 Nov 15 15:05 .env.testing-ci
-rw-r--r--  1 apache apache    554 Nov 15 15:05 .env.tests
--w-------  1 apache apache   1711 Nov 15 15:05 gh-changelog.php
drwxr-xr-x  8 apache apache   4096 Nov 16 14:10 .git
-rwxr-xr-x  1 apache apache     11 Nov 15 15:05 .gitattributes
drwxr-xr-x  2 apache apache   4096 Nov 15 15:05 .github
-rwxr-xr-x  1 apache apache   1061 Nov 15 15:05 .gitignore
-rw-r--r--  1 apache apache    251 Nov 15 15:05 .htaccess
--w-------  1 apache apache    956 Nov 15 15:05 index.html
--wx--x--x  1 apache apache    371 Nov 15 15:05 install.sh
--wx--x--x  1 apache apache  34519 Nov 15 15:05 LICENSE
--w-------  1 apache apache   1686 Nov 15 15:05 package.json
--w-------  1 apache apache    934 Nov 15 15:05 phpunit.xml
d-wx--x--x  8 apache apache   4096 Nov 15 15:05 public
--w-------  1 apache apache  25646 Nov 15 15:05 README.md
d-wx--x--x  6 apache apache   4096 Nov 15 15:05 resources
d-wx--x--x  3 apache apache   4096 Nov 15 15:05 routes
d-wx--x--x  2 apache apache   4096 Nov 15 15:05 sample_csvs
--w-------  1 apache apache    567 Nov 15 15:05 server.php
--w-------  1 apache apache  24622 Nov 15 15:05 snipeit.sh
drwxr-xr-x  7 apache apache   4096 Nov 15 15:05 storage
d-wx--x--x 10 apache apache   4096 Nov 15 15:05 tests
-rw-r--r--  1 apache apache   2379 Nov 15 15:05 .travis.yml
--w-------  1 apache apache   6083 Nov 15 15:05 upgrade.php
d-wx--x--x 51 apache apache   4096 Nov 16 14:14 vendor
--w-------  1 apache apache   2130 Nov 15 15:05 webpack.mix.js
```

Looking for some help. Thanks.

**Comment by snipe** [comment]
Can you give me the output from your log? That error can be a little misleading, as it usually means something bonked really early on in the framework loading. We sometimes see it if there are extra spaces in the .env, or if something that should be enclosed in quotes in the .env isn't. It's *usually* some .env weirdness, but can be super hard to track down.

**Comment by snipe** [comment]
(Note: there may not be anything in your log, since it's the log call that it's choking on)

**Comment by m384wwn** [comment]
Yeah, nothing in the logs:

[root@web05 logs]# pwd
/var/www/snipe-it/storage/logs
[root@web05 logs]# ll
total 0

**Comment by snipe** [comment]
What's the permission on your logs dir?

**Comment by m384wwn** [comment]
[root@web05 logs]# ll /var/www/snipe-it/storage/
total 20
drwxr-xr-x 3 apache apache 4096 Nov 15 15:05 app
drwxr-xr-x 2 apache apache 4096 Nov 15 15:05 debugbar
drwxr-xr-x 5 apache apache 4096 Nov 15 15:05 framework
**drwxr-xr-x 2 apache apache 4096 Nov 15 15:05 logs**
drwxr-xr-x 8 apache apache 4096 Nov 15 15:05 private_uploads

**Comment by snipe** [comment]
Blergh. Anything in your php error logs pr apache error logs?

**Comment by m384wwn** [comment]
Just that one error:


[root@web05 httpd]# cat /var/log/httpd/* | grep error
[Thu Nov 16 18:53:02 2017] [error] [client 10.128.6.66] PHP Fatal error:  Class 'Log' not found in /var/www/snipe-it/app/Exceptions/Handler.php on line 39

**Comment by snipe** [comment]
Okay this is going to sound weird, but we need to eliminate the .env from the equation, since when I see this Log message, it *usually* relates to an .env setting.

- Clear your config cache (if you can): `php artisan cache:clear`
- Backup your current env: `cp .env .env.backup` 
- Then starting at the bottom, start deleting line by line and refreshing the web UI. You'll need to put all those things back, but if there's a line in there that it doesn't like, this will help us identify it - or at the very least, rule it out as a possible cause.

**Comment by m384wwn** [comment]
I get what you are saying but my feeling was that it might be from something related to the composer install / composer dependency install. I messed it up something good originally and had to correct it.

So i blew away the /var/www/snipe-it/ directory and started again.

Aside from having the wrong DB_CHARSET it worked!! I copied back the .env from this thread - changed AP__KEY - and it also worked. 

Thank you for helping. :)
